topsites.py

```diff
@@ -0,0 +1,180 @@
+            dom = parseString(response.content)
+            sites = dom.getElementsByTagName("aws:Site")
+            for site in sites:
+                url = self.get_text(
+                    site.getElementsByTagName("aws:DataUrl")[0].childNodes)
+                rank = int(self.get_text(
+                    site.getElementsByTagName("aws:Rank")[0].childNodes))
+
+                priority = begin_priority
+                if rank > 100 and rank <= 1000:
+                    priority = priority + 1
+                elif rank > 1000 and rank <= 10000:
+                    priority = priority + 2
+
+                if url in topsites:
+                    site_row = topsites[url]
+                    # Update priority is new is lower
+                    if site_row.priority > priority:
+                        site_row.priority = priority
+                        site_row.country_code = self.country_code
+                        site_row.ranking = rank
+                else:
+                    site_row = Site(url, priority, self.country_code, rank)
+                    topsites[url] = site_row
+                    session.add(site_row)
+
+            session.commit()
+
+    def build_query(self, start, count):
+        '''Build query string for request with start ranking and count'''
+        now = time.time()
+        mlsec = repr(now).split('.')[1][:3]
+        nowtime = datetime.datetime.utcnow()
+        timestamp = nowtime.strftime(ATS_DATEFORMAT).format(mlsec)
+
+        query_params = {
+            "Action": ATS_ACTION_NAME,
+            "ResponseGroup": ATS_RESPONSE_GROUP_NAME,
+            "AWSAccessKeyId": self.access_key,
+            "Timestamp": timestamp,
+            "CountryCode": self.country_code,
+            "Count": count,
+            "Start": start,
+            "SignatureVersion": "2",
+            "SignatureMethod": ATS_HASH_ALGORITHM
+        }
+
+        # alexa top site only accept request with ordered query parameters
+        return urlencode(collections.OrderedDict(sorted(query_params.items())))
+
+    def gen_sign(self, data, secret_key):
+        '''Computes RFC 2104-compliant HMAC signature'''
+        dig = hmac.new(secret_key, data, hashlib.sha256).digest()
+        return base64.b64encode(dig)
+
+    def get_text(self, node_list):
+        '''Get text from text node'''
+        rc = []
+        for node in node_list:
+            if node.nodeType == node.TEXT_NODE:
+                rc.append(node.data)
+        return ''.join(rc)
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 3:
+        print("Usage: python topsites.py ACCESS_KEY_ID SECRET_ACCESS_KEY")
+        sys.exit(1)
+
+    access_key = sys.argv[1]
+    secret_key = sys.argv[2]
+
+    site = TopSites(access_key, secret_key)
+
+    for country_code in REGIONS:
+        site.country_code = country_code
+        end_ranking = REGIONS[country_code][0]
+        begin_priority = REGIONS[country_code][1]
```

Comment:
```python
end_ranking, begin_priority = REGIONS[country_code]
```
will work by assigning the two values of the tuple directly to each of the variable. One line instead of two without using indexes.